src/types.ts

```diff
@@ -2155,19 +2158,11 @@ export class ZodArray<
     return this.min(1, message) as any;
   }
 
-  unique(
-    params: {
-      identifier?: ArrayUniqueIdentifier<output<T>>;
-      message?:
-        | errorUtil.ErrMessage
-        | ((duplicateElements: Array<output<T>>) => string);
-      showDuplicates?: boolean;
-    } = {}
-  ): this {
+  unique(params: ZodArrayDef<T>["uniqueness"] = {}): this {
     const message =
-      typeof params.message === "function"
+      typeof params?.message === "function"
         ? params.message
-        : errorUtil.toString(params.message);
+        : errorUtil.toString(params?.message);
```

Comment:
```suggestion
        : errorUtil.toString(params.message);
```

Reply:
I need to put this `?` because the params can be not passed (`undefined`) on the `.unique` method.

Reply:
My bad